deeplearning/src/main/java/nl/tudelft/bep/deeplearning/clustering/Cluster.java

```diff
@@ -0,0 +1,146 @@
+package nl.tudelft.bep.deeplearning.clustering;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+public class Cluster implements Comparable<Cluster> {
+
+	//Coordinates of the point or in the case of the cluster the mean.
+	private double x, y;
+	private int id;
+	private ArrayList<Cluster> set;
+
+	public Cluster(double x, double y, int id) {
+		this.x = x;
```

Comment:
The body of this constructor could be replaced with:
this(x,y,id,new ArrayList<>());